apis/projectcontour/v1alpha1/contourdeployment.go

```diff
@@ -122,6 +122,11 @@ type ContourSettings struct {
 	// the annotations for Prometheus will be appended or overwritten with predefined value.
 	// +optional
 	PodAnnotations map[string]string `json:"podAnnotations,omitempty"`
+
+	// WatchNamespaces is an array of namespaces. Setting it will instruct the contour instance
+	// to only watch this subset of namespaces.
+	// +optional
+	WatchNamespaces []string `json:"watchNamespaces,omitempty"`
```

Comment:
hm, lets add a kubebuilder min/max length here so we dont get any empty lists or allow a huge list

we should probably also add some validation that we're not getting anything that is an invalid namespace name (i.e. not empty, longer than the maximum ns length, or with invalid characters) since we'll be generating resources based on what is provided

heres an example of proper validation from gw api: https://github.com/kubernetes-sigs/gateway-api/blob/54387b1e8db24e0c3eb644fa6c48a1b9ee711b39/apis/v1/shared_types.go#L602-L605

we've been pretty inconsistent about this sort of validation

Reply:
Do you think we should make sure the namespace exist when creating the ContourDeployment? Or just when creating corresponding contour Object. I prefer the second one.
But for basic validation, I agree it make sense to put it inside kubebuilder validation.

Reply:
Ok looks like it's still impossible to validate for []string type using annotation
https://github.com/kubernetes-sigs/controller-tools/issues/342

Reply:
Then do we show the invalid status on the Gateway? Since I don't see a controller just for ContourDeployment. 
I guess it make sense not to have a specific ContourDeployment Controller since it's just a config.

Reply:
Maybe add a validation webhook of ContourDeployment for that case?